### ISSUE TITLE
Support for IAsyncEnumerable<T> in list calls

### ISSUE THREAD

**Issue by WhitWaldo** [Azure/azure-libraries-for-net#1038]
Presently, when listing a given resource with this API, I can either perform the call synchronously and receive an IEnumerable<T> or I can perform the call asychronously, but I received an IPagedCollection<T> that I'm responsible for enumerating and performing subsequent asynchronous calls against. This adds several more boilerplate lines just to implement asynchronous functionality. It's not overbearing, but it could be easier.

C# 8.0 introduced the idea of async streams with the IAsyncEnumerable<T>. Per the [documentation](https://docs.microsoft.com/en-us/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8), if implemented, enumerating the values of a collection could be simplified to more closely resemble that of the synchronous IEnumerable:

var eventHubs = azure.EventHubs.ListByNamespaceAsync(_resourceGroupName, _namespaceName);
await foreach (var item in eventHubs)
  Console.WriteLine(item.Name);

Put simply, the asynchronous result could be iterated simply by putting an await in front of the foreach.

For those users that have already implemented using IPagedCollection, this could simply be introduced as another overload to each of the ListAsync method enabling those that want to simplify their code to do so, leaving no one with a need to necessarily migrate existing logic.

Thank you for your consideration.

**Comment by nickzhums** [comment]
@WhitWaldo Hi Whit, thanks for reporting your issue. We will discuss about this :)

**Comment by WhitWaldo** [comment]
@yungezz I noticed you added the Event Hubs label. I use that as just one example - just about every Azure resource has a ListAsync method.

**Comment by nickzhums** [comment]
@WhitWaldo we took a look at this and this may require quite some work as it involves re-writing all the method signatures. That being said, your need might be solved when Track 2 .NET Management SDK comes out
